### PR TITLE
SNOW-280651 Add examples for key-pair authentication and OAuth to the README file

### PR DESCRIPTION
**Pull request by sfc-gh-dyoshinaga**
Description
Add examples for key-pair authentication and OAuth to the README file

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -141,6 +141,87 @@ using (IDbConnection conn = new SnowflakeDbConnection())
...
+    * `{passwordForDecryptingThePrivateKey}` is the password for decrypting the private key.
+
+  * Specify an unencrypted private key (read from a file):
+
+    ```cs
+    using (IDbConnection conn = new SnowflakeDbConnection())
+    {
+        string privateKeyContent = File.ReadAllText({pathToThePrivateKeyFile}).Replace("=", "==");
+
+        conn.ConnectionString = String.Format("account=testaccount;authenticator=snowflake_jwt;user=testuser;private_key={0}", privateKeyContent);
```
Comment (SimbaGithub) [inline README.md:193]: db=testdb;schema=testschema can be added if you want to use the same set of properties for all connection strings.

**Comment by SimbaGithub** [comment]
Just a question: Should examples for the existing external-browser and okta authentication mechanisms be in the README to document all the mechanisms the connector supports?

File: README.md (hunk 1 above)
Comment (sfc-gh-dyoshinaga) [reply README.md]: Thanks! I added this. I also made the order of the parameters in the example more consistent.

**Comment by sfc-gh-dyoshinaga** [comment]
> Just a question: Should examples for the existing external-browser and okta authentication mechanisms be in the README to document all the mechanisms the connector supports?

Thanks! I added examples of externalbrowser and native SSO through Okta.

**Comment by SimbaGithub** [comment]
LGTM :)

**Review by sfc-gh-hyu — APPROVED** [review]

**Pull request merged**